code/datums/abilities/wraith.dm

```diff
@@ -723,6 +860,1075 @@
+			ability.doCooldown()
+			ability = holder.getAbility(/datum/targetable/wraithAbility/curse/death)
+			ability.doCooldown()
+
+			var/image/icon = image('icons/mob/wraith_ui.dmi', icon_state = "weak_status", loc = H)
+			icon.blend_mode = BLEND_ADD
+			icon.plane = PLANE_ABOVE_LIGHTING
+			icon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			icon.pixel_y = 28
+			icon.alpha = 170
+			get_image_group(CLIENT_IMAGE_GROUP_CURSES).add_image(icon)
+			H.curse_icons += icon
+
+			return 0
+		else
+			return 1
+
+/datum/targetable/wraithAbility/curse/rot
+	name = "Curse of rot"
+	icon_state = "rotcurse"
+	desc = "Curse the living with a netherworldly plague."
+	targeted = 1
+	pointCost = 40
+	cooldown = 45 SECONDS
+
+	cast(atom/target)
+		if (..())
+			return 1
+
+		if (ishuman(target))
+			var/mob/living/carbon/human/H= target
+			if(H.bioHolder.HasEffect("rot_curse"))
+				boutput(holder.owner, "That curse is already applied to this being...")
+				return 1
+			usr.playsound_local(usr.loc, "sound/voice/wraith/wraithspook[rand(1, 2)].ogg", 80, 0)
+			H.bioHolder.AddEffect("rot_curse")
+			boutput(holder.owner, "<span class='notice'>We curse this being with a decaying curse.</span>")
+			var/datum/targetable/ability = holder.getAbility(/datum/targetable/wraithAbility/curse/blood)
+			ability.doCooldown()
+			ability = holder.getAbility(/datum/targetable/wraithAbility/curse/blindness)
+			ability.doCooldown()
+			ability = holder.getAbility(/datum/targetable/wraithAbility/curse/enfeeble)
+			ability.doCooldown()
+			ability = holder.getAbility(/datum/targetable/wraithAbility/curse/death)
+			ability.doCooldown()
+
+			var/image/icon = image('icons/mob/wraith_ui.dmi', icon_state = "rot_status", loc = H)
+			icon.blend_mode = BLEND_ADD
+			icon.plane = PLANE_ABOVE_LIGHTING
+			icon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			icon.pixel_y = 26
+			icon.alpha = 170
+			get_image_group(CLIENT_IMAGE_GROUP_CURSES).add_image(icon)
+			H.curse_icons += icon
+
+			return 0
+		else
+			return 1
+
+/datum/targetable/wraithAbility/curse/death	//Only castable if you already put 4 curses on someone
+	name = "Curse of death"
+	icon_state = "deathcurse"
+	desc = "Reap a fully cursed being's soul!"
+	targeted = 1
+	pointCost = 80
+	cooldown = 45 SECONDS
+
+	cast(atom/target)
+		if (..())
+			return 1
+
+		if (ishuman(target))
+			var/mob/living/carbon/human/H = target
+			var/mob/wraith/W = holder.owner
+			if (H?.bioHolder.HasEffect("rot_curse") && H?.bioHolder.HasEffect("weak_curse") && H?.bioHolder.HasEffect("blind_curse") && H?.bioHolder.HasEffect("blood_curse"))
+				W.playsound_local(W.loc, "sound/voice/wraith/wraithhaunt.ogg", 80, 0)
+				boutput(holder.owner, "<span class='alert'>That soul is OURS!!</span>")
+				boutput(H, "<span class='alert'>The voices in your heads are reaching a crescendo!</span>")
+				H.make_jittery(300)
+				sleep(4 SECOND)
```

Comment:
I'd be wary of sleeping in `cast` without a spawn, will probably break something with ability code

Reply:
I've switched all the "sleep" into "SPAWN", not sure if this is the way